bin/mmdb_utils.py

```diff
@@ -25,7 +25,7 @@
 LIMITS_CONF_STANZA = 'iplocation'
 LIMITS_CONF_PARAMETER = 'db_path'
 
-MaxMindDatabaseDownloadLink = 'https://updates.maxmind.com/geoip/databases/GeoLite2-City/update?db_md5=00000000000000000000000000000000'
+MaxMindDatabaseDownloadLink = 'https://download.maxmind.com/geoip/databases/GeoLite2-City/download?suffix=tar.gz'
```

Comment:
We moved from download to update to have basic auth for authentication. Is basic auth supported on this endpoint?

Reply:
MaxMindDatabaseDownloadLink = 'https://updates.maxmind.com/geoip/databases/GeoLite2-City/update?db_md5=00000000000000000000000000000000'
This link don't redirect and gives direct response

MaxMindDatabaseDownloadLink = 'https://download.maxmind.com/geoip/databases/GeoLite2-City/download?suffix=tar.gz'
Redirects to work -                    mm-prod-geoip-databases.a2649acb697e2c09b632799562c076f2.r2.cloudflarestorage.com

MaxMindDatabaseDownloadLink = 'https://mm-prod-geoip-databases.a2649acb697e2c09b632799562c076f2.r2.cloudflarestorage.com/geoip/databases/GeoLite2-City/download?suffix=tar.gz'
don't work direct link

Reference - https://dev.maxmind.com/geoip/updating-databases?utm_campaign=R2%20presigned%20URLs&utm_medium=email&_hsmi=297797011&_hsenc=p2ANqtz-9qbZQb5dJ3qB_OcpYW8cwAre8Sr6msAhl2SdR2gP_oACzJpveuy0_gjXwb09C_8eWKA0nrccs-R-k0ORO_Yca7TQDdxw&utm_content=297797011&utm_source=hs_email#directly-downloading-databases
Reference - Download link copied from here.

## This code works

```
MMDB_PATH_DIR = 'mmdb'
MMDB_FILE_NAME = 'GeoLite2-City.mmdb'

ACCEPTED_LOOKUP_NAME = 'GeoIP2-City.mmdb'


DB_DIR_TEMP_PATH = os.path.join(os.path.dirname(__file__), MMDB_PATH_DIR)
DB_TEMP_DOWNLOAD = os.path.join(DB_DIR_TEMP_PATH, "temp_file.gz")



def download_mmdb_database(account_id, license_key):
    # NOTE - Proxy Configuration
    # Remove '<username>:<password>@' part if using proxy without authentication (just use ip:port format)
    # Understand the risk of storing password in plain-text when using proxy with authentication
    proxies = None
    '''
    proxies = {
        "http" : "<proxy-supported-schema http|https>://<username>:<password>@<ip-address>:<port>",
        "https" : "<proxy-supported-schema http|https>://<username>:<password>@<ip-address>:<port>"
    }
    '''

    # NOTE - Please visit GitHub page (https://github.com/VatsalJagani/Splunk-App-Auto-Update-MaxMind-Database), if you are developer and want to help improving this App in anyways

    print("Downloading the MaxMind DB file.")
    try:
        r = requests.get(MaxMindDatabaseDownloadLink, auth=(account_id, license_key), allow_redirects=True, proxies=proxies)
        print('Redirect URL:', r.url)

    except Exception as err:
        print("Failed to download MaxMind DB file from {}".format(MaxMindDatabaseDownloadLink))
        raise err

    if r.status_code == 200:
        with open(DB_TEMP_DOWNLOAD, 'wb') as fp:
            fp.write(r.content)

        downloaded_file = os.path.join(DB_DIR_TEMP_PATH, 'GeoLite2-City.mmdb')
        try:
            # Extract the downloaded file
            with gzip.open(DB_TEMP_DOWNLOAD, 'rb') as s_file, open(downloaded_file, 'wb') as d_file:
                shutil.copyfileobj(s_file, d_file, 65536)
        except Exception as e:
            msg = "Unable to extract downloaded MaxMind database. {}".format(e)
            print(msg)
            raise e

        print("Downloaded MaxMind DB file: {}".format(downloaded_file))
    else:
        err_msg = "Unable to download Max Mind database. status_code={}, Content: {}".format(r.status_code, r.content)
        print(err_msg)
        raise Exception(err_msg)



print("Initialized MaxMindDatabaseUtil")

account_id = "4****"
license_key = "Rk***"
download_mmdb_database(account_id, license_key)

```

Reply:
Okay. The geoipupdate utility also uses the updates.maxmind.com. so I don't think we need to change anything for now.

Reply:
The change is only for download endpoint. right? why are we changing the endpoint?

Reply:
Yes, that's the new endpoint if we use that endpoint it will redirect to new r2 endpoint as per the doc. And we are able to successfully download the file.